python/cuml/benchmark/bench_runners.py

```diff
@@ -0,0 +1,143 @@
+                    cpu_time=cpu_elapsed,
+                    speedup=cpu_elapsed / float(cu_elapsed),
+                    n_samples=n_samples,
+                    n_features=n_features,
+                    **param_overrides,
+                    **cuml_param_overrides)
+
+    def run(self, algo_pair, param_overrides={},
+            cuml_param_overrides={},
+            cpu_param_overrides={}, *, run_cpu=True):
+        all_results = []
+        for ns in self.bench_rows:
+            for nf in self.bench_dims:
+                try:
+                    all_results.append(self._run_one_size(algo_pair,
+                                                          ns,
+                                                          nf,
+                                                          param_overrides,
+                                                          cuml_param_overrides,
+                                                          cpu_param_overrides, run_cpu))
+                except Exception as e:
+                    print("Failed to run with %d samples, %d features: %s" %
+                          (ns, nf, str(e)))
+                    raise
+                    all_results.append(dict(n_samples=ns, n_features=nf))
+        return all_results
+
+
+
+class AccuracyComparisonRunner(SpeedupComparisonRunner):
+    """Wrapper to run an algorithm with multiple dataset sizes
+    and compute accuracy and speedup of cuml relative to sklearn
+    baseline."""
+    def __init__(self, bench_rows, bench_dims,
+                 dataset_name='blobs', input_type='numpy',
+                 test_fraction=0.10):
+        super().__init__(bench_rows, bench_dims, dataset_name, input_type)
+        self.test_fraction = 0.10
+
+    def _run_one_size(self, algo_pair, n_samples, n_features,
+                      param_overrides={},
+                      cuml_param_overrides={},
+                      cpu_param_overrides={},
+                      run_cpu=True):
+        # keep training set size constant even as we add test
+        n_samples_with_test = int(n_samples / (1 - self.test_fraction))
+
+        data = bench_data.gen_data(self.dataset_name,
+                                   self.input_type,
+                                   n_samples_with_test,
+                                   n_features,
+                                   test_fraction=self.test_fraction)
+        X_test, y_test = data[2:]
+
+        cu_start = time.time()
+        cuml_model = algo_pair.run_cuml(data, **param_overrides, **cuml_param_overrides)
+        cu_elapsed = time.time() - cu_start
+
+        if algo_pair.accuracy_function:
+            if hasattr(cuml_model, 'predict'):
+                y_pred_cuml = cuml_model.predict(X_test)
+            else:
+                y_pred_cuml = cuml_model.transform(X_test)
+            cuml_accuracy = algo_pair.accuracy_function(y_test, np.asarray(y_pred_cuml))
+        else:
+            cuml_accuracy = 0.0
```

Comment:
Would it make sense to just compare the outputs directly in the case no accuracy function has been provided? I'm thinking of NearestNeighbors here. Or maybe to compare the results with some predefined tolerance?